@app/server/src/middleware/installCors.ts

```diff
@@ -0,0 +1,19 @@
+import cors from "cors";
+import { Express } from "express";
+
+export default (app: Express) => {
+  const origin = [];
+  if (process.env.ROOT_URL) {
+    origin.push(process.env.ROOT_URL);
+  }
+  if (process.env.CORS_ALLOWED_URLS) {
+    origin.push(
+      ...(process.env.CORS_ALLOWED_URLS?.replace(/s\s/g, "").split(",") || [])
+    );
+  }
+  const corsOptions = {
+    origin,
+    credentials: true,
+  };
+  app.use(cors(corsOptions));
```

Comment:
I'm not sure your regexp `/s\s/g` is correct... Why is the literal `s` in there? Also I think we could compress this down a bit, improving the TypeScript inference/error messages without making it harder to read:

```suggestion
  app.use(cors({
    origin: [
      ...(process.env.ROOT_URL ? [process.env.ROOT_URL] : []),
      ...(process.env.CORS_ALLOWED_URLS?.replace(/\s+/g, "").split(",") || [])
    ],
    credentials: true,
  }}));
```